tests/image-stream/dmg.spec.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const m = require('mochainon');
+const fs = require('fs');
+const path = require('path');
+const DATA_PATH = path.join(__dirname, 'data');
+const IMAGES_PATH = path.join(DATA_PATH, 'images');
+const DMG_PATH = path.join(DATA_PATH, 'dmg');
+const imageStream = require('../../lib/image-stream/index');
+const tester = require('./tester');
+
+describe('ImageStream: DMG', function() {
+
+  this.timeout(20000);
+
+  describe('.getFromFilePath()', function() {
+
+    describe('given an dmg image', function() {
+      tester.extractFromFilePath(
+        path.join(DMG_PATH, 'zlib-compressed.dmg'),
+        path.join(IMAGES_PATH, 'zlib-compressed.img'));
+    });
+
+  });
+
+  describe('.getImageMetadata()', function() {
+
+    it('should return the correct metadata', function() {
+      const image = path.join(DMG_PATH, 'zlib-compressed.dmg');
```

Comment:
I've got a range of test images in the `udif` module here: https://github.com/jhermsmeier/node-udif/tree/master/test/data Not sure we'd need to test the same stuff in here, as that's internal stuff Etcher doesn't interface with. WDYT?